src/mme-app/actionHandlers/attachActionHandlers.cpp

```diff
@@ -750,11 +856,71 @@ ActStatus ActionHandlers::cs_req_to_sgw(SM::ControlBlock& cb)
 	const DigitRegister15& ueMSISDN = ue_ctxt->getMsisdn();
 	ueMSISDN.convertToBcdArray(cs_msg.MSISDN);
 
-	cmn::ipc::IpcAddress destAddr;
-	destAddr.u32 = TipcServiceInstance::s11AppInstanceNum_c;
+	//cmn::ipc::IpcAddress destAddr;
+	destAddr_dns.u32 = TipcServiceInstance::s11AppInstanceNum_c;
+	if(g_mme_cfg.dns_config.dns_flag == 1)	
+	{
 
+		SetDNSConfiguration();
+		/* Query DNS based on lb and hb of tac */
+		char lb[LB_HB_LEN] = {0};
+		char hb[LB_HB_LEN] = {0};
+		char mnc[MCC_MNC_LEN] = {0};
+		char mcc[MCC_MNC_LEN] = {0};
+		short tac=0;
+		unsigned long sgw_ip = 0;
+		char mccDigit1 =  ue_ctxt->getTai().tai_m.plmn_id.idx[0] & 0x0F;
+		char mccDigit2 = (ue_ctxt->getTai().tai_m.plmn_id.idx[0] & 0xF0) >> 4;
+		char mccDigit3 = ue_ctxt->getTai().tai_m.plmn_id.idx[1] & 0x0F;
+		char mncDigit1 = ue_ctxt->getTai().tai_m.plmn_id.idx[2] & 0x0F;
+		char mncDigit2 = (ue_ctxt->getTai().tai_m.plmn_id.idx[2] & 0xF0) >> 4;
+		char mncDigit3 = (ue_ctxt->getTai().tai_m.plmn_id.idx[1] & 0xF0) >> 4;
+
+
+
+		if (mncDigit3 == 15)
+			snprintf(mnc, MCC_MNC_LEN, "%u%u", mncDigit1,mncDigit2);
+		else
+			snprintf(mnc, MCC_MNC_LEN, "%u%u%u", mncDigit1,mncDigit2,mncDigit3);
+
+
+		snprintf(mcc,MCC_MNC_LEN, "%u%u%u", mccDigit1, mccDigit2, mccDigit3);
+
+		log_msg(LOG_ERROR,"%u %u  %u %u %u %u mcc and mnc digits\n", mccDigit1, mccDigit2, mccDigit2, mncDigit1, mncDigit2, mncDigit3);
+		log_msg(LOG_ERROR, "%s %s: mnc and mcc\n", mnc,mcc);
+		memcpy(plmn_id, ue_ctxt->getTai().tai_m.plmn_id.idx, 3);
+		log_msg(LOG_ERROR, "%s %s: plmn_id\n", plmn_id,ue_ctxt->getTai().tai_m.plmn_id.idx);
+		tac = ntohs(ue_ctxt->getTai().tai_m.tac);
+		log_msg(LOG_ERROR, "%d %d tac value* \n",tac, ue_ctxt->getTai().tai_m.tac);
+
+		if ( tac != 1) {
```

Comment:
remove this code. Why tac has to be 1 always ? It might be true in development environment...better to add code under DEV_TEST macro and delete this code before sending review.